### PR TITLE
Update comparison-to-olm.md

### PR DESCRIPTION
**Pull request by cdjohnson**
Add some additional clarification and updates based on some new OLM features.

**What this PR does / why we need it**:
Add clarity over some of the features that ODLM provides and added some updates to some of the newer features available in OLM.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #
NA

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by horis233 — APPROVED** [review]
/lgtm

File: docs/design/comparison-to-olm.md [hunk 1]
```diff
@@ -45,12 +45,21 @@ The ODLM manages the deployment of the operands for OLM managed operators.  This
...
+- **OLM** creates all required, dependent operators automatically by creating additional Subscriptions automatically.  Each operator statically defines all of it's REQUIRED dependencies in the form of CustomResourceDefinitions, API Service Definitions or Operator Package versions.  
+  - **OLM** operators declare which CRDs, and API Services are PROVIDED.
+  - **OLM** operators declare which CRDs, API Services and other Operator Package Versions are REQUIRED.
+  - All dependencies are statically defined as REQUIRED (vs. optional, or preferred), which result in all dependencies must be installed as a unit.
+  - All dependencies must be in the same OperatorGroup, which may require the operators to be in the same namespace. 
+  - Example:  The IAM Operator REQUIRES the Mongo DB CRD and the Mongo DB Operator PROVIDES the Mongo DB CRD.  When the IAM Operator Subscription is created, the Mongo DB Subscription is dynamically created, satisfying the Mongo DB CRD dependency.
```
Comment (pgodowski) [inline docs/design/comparison-to-olm.md:57]: Can we not use the IAM and Mongo in examples?

Comment (horis233) [reply docs/design/comparison-to-olm.md:57]: @pgodowski 

Do you mean we should use more general name, like `bar operator` and `foo operator`?

Comment (pgodowski) [reply docs/design/comparison-to-olm.md:57]: @horis233 yes, perhaps a `parent` and `child` or something similar like requestor-requestee, etc

Comment (cdjohnson) [reply docs/design/comparison-to-olm.md:57]: Yeah, I agree... that would add some additional clarity.

Comment (cdjohnson) [reply docs/design/comparison-to-olm.md:57]: Changes in https://github.com/IBM/operand-deployment-lifecycle-manager/pull/634

**Pull request merged**